crates/rspack_plugin_javascript/src/visitors/dependency/context_dependency_helper.rs

```diff
@@ -190,3 +220,23 @@ pub fn create_context_dependency(
     }
   }
 }
+
+struct BasicEvaluatedExpressionVisitor<'a, F: FnMut(&Expr)> {
+  targets: Vec<&'a BasicEvaluatedExpression>,
+  on_visit: F,
+}
+
+impl<'a, F: FnMut(&Expr)> Visit for BasicEvaluatedExpressionVisitor<'a, F> {
+  fn visit_expr(&mut self, n: &Expr) {
```

Comment:
Walk is different with swc visitor, parser plugins' goal is replace all the swc visitors, so this needs to rewrite to walk_expression

Reply:
🤔 I use this visitor to find the `Expr` ast node corresponding to the BasicEvaluatedExpression and then call `parser.walk_expression`. how would this visitor be rewritten to `walk_expression`?

Reply:
I think there could have some walk around to implement this and avoid the life time issue, I will take a look next week

Reply:
I decide to not implement `setExpression` for BasicEvaluatedExpression, which will introduce lots of lifetime for the parser, basically only context dependency is using `setExpression`, which compare to the current solution it has no different in most cases, and in some cases, it will have slight better performance, but I think it's acceptable